src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisState.SyntaxReferenceAnalyzerStateData.cs

```diff
@@ -66,6 +73,34 @@ public override void Free()
         }
 
         /// <summary>
+        /// Stores the partial analysis state for operation actions executed on the declaration.
+        /// </summary>
+        internal class OperationAnalyzerStateData : AnalyzerStateData
+        {
+            public HashSet<IOperation> ProcessedOperations { get; set; }
```

Comment:
Why a HashSet? Are we guaranteeing that operations returned from two calls to the same method are Equals?